concepts/eventing/backend-crd.md

```diff
@@ -0,0 +1,129 @@
+    type: Eventing Backend Ready
+  - lastTransitionTime: "2022-05-06T08:35:11Z"
+    reason: Publisher proxy deployment ready
+    status: "True"
+    type: Publisher Proxy Ready
+  - lastTransitionTime: "2022-05-06T08:35:11Z"
+    reason: Subscription controller started
+    status: "True"
+    type: Subscription Controller Ready
+  eventingReady: true
+```
+
+> **Note:** As a result of having a flexible design for the backend configuration in the CRD, we decided not to mention which configuration needs to move from the Eventing controller charts to the backend CR and left it to the implementation phase.
+
+## Metadata
+
+The `namespaces` and `name` can be set by users. This allows to run multiple different instances of the same backend type.
+
+```yaml
+metadata:
+  name: any
+  namespace: any
+  labels:
+    kyma-project.io/eventing: backend
+```
+
+## Spec
+
+Each backend instance is defined with its own `spec`. The `spec` contains the objects `type` and `config`. While `type` accepts the string of the name of the backend (`"nats"`, `"beb"`, ...etc), the `config` accepts the key/value pairs of the actual configuration for the specified backend.
+
+Example how to configure `BEB` as the Eventing backend:
+
+```yaml
+spec:
+  type: "beb"
+  config:
+  - secretRef: "some-secret"
+```
+
+## Validation
+
+- Spec config validation can be done by writing a custom `ValidatingAdmissionWebhook`.
+- Defaulting the required but not-set spec config can be done by writing a custom `MutatingAdmissionWebhook`.
+
+The following scenarios should return a `validationError` and result into the backend not being created.
+- Spec config keys that do not belong to the specified backend.
+- Spec config values set to forbidden data.
+- Spec config values using the wrong data type for the given key.
+  Similar, changing the specs of an existing backend in an invalid way should be rejected and return a `validationError`.
+
+## Status
+
+The `status` contains the following fields:
+- `conditions` contains the readiness of all the components which the backend depends on.
+- `eventingReady` is evaluated by `ANDing` all the `status.conditions`.
+
+Example of a backend CR with no validation errors:
+
+```yaml
+status:
+  conditions:
+  - lastTransitionTime: "2022-05-06T08:35:11Z"
+    reason: Eventing Backend Ready
+    status: "True"
+    type: Eventing Backend Ready
+  - lastTransitionTime: "2022-05-06T08:35:11Z"
+    reason: Publisher proxy deployment ready
+    status: "True"
+    type: Publisher Proxy Ready
+  - lastTransitionTime: "2022-05-06T08:35:11Z"
+    reason: Subscription controller started
+    status: "True"
+    type: Subscription Controller Ready
+  eventingReady: true
+```
+
+> **Note:** The `Eventing Backend Ready` condition should indicate if the configured backend type in the spec is ready for use or not. It can be as simple as checking the underlying connection status with the backend. But that decision is left to the implementation phase.
```

Comment:
```suggestion
> **Note:** The `Eventing Backend Ready` condition indicates if the configured backend type in the spec is ready for use. It can be as simple as checking the underlying connection status with the backend. This decision is left for the implementation phase.
```